### PR TITLE
indication of number of rows based on pagination

### PR DESCRIPTION
**Pull request by wajiha-oicr**
**Description of changes**

- optional prop added to table, to display a "rows x - y out of z" above the table
- Memoized function calculates said value based on the page index + number of rows visible
- Added jsx element as a prop so the row count indicator can align with other elements

**Type of Change**

- [ ] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- [ ] Matches design:

  - component sizes, spacing, and styles
  - font size, weight, colour
  - spelling has been double checked

- [ ] New uikit components have Storybook stories
- [ ] Feature is minimally responsive.
- [ ] Manual testing of UI feature.
- [ ] Selenium test is completed and passing.

### CODE REVIEW

File: uikit/Table/index.tsx [hunk 1]
```diff
@@ -117,39 +124,81 @@ function Table<Data extends TableDataBase>({
...
-      PaginationComponent={PaginationComponent}
-      NoDataComponent={NoDataComponent}
-      showPagination={isEmpty(data) ? false : showPagination}
-      getNoDataProps={x => x}
-      sortable={sortable}
-      resizable={resizable}
-      {...rest}
-    />
+    <>
+      <div
```
Comment (wajiha-oicr) [inline uikit/Table/index.tsx:152]: this div is new

File: uikit/Table/index.tsx [hunk 2]
```diff
@@ -117,39 +124,81 @@ function Table<Data extends TableDataBase>({
...
+        {data.length > 0 && rowName && PageRowCounter()}
+        {tableTopContent}
+      </div>
+      <StyledTable
+        style={{
+          // this is written with style object because css prop somehow only applies to the header
+          transition: 'all 0.25s',
+          filter: resizing && withResizeBlur ? 'blur(8px)' : 'blur(0px)',
+          width,
+        }}
```
Comment (wajiha-oicr) [inline uikit/Table/index.tsx:169]: this is the other change here, everything else is just a margin diff

File: components/pages/file-repository/FileTable/index.tsx [hunk 3]
```diff
@@ -144,16 +144,91 @@ export default ({
...
+        display: flex;
+      `}
+    >
+      <DropdownButton
+        css={css`
+          margin-right: 8px;
+        `}
+        variant="secondary"
+        size="sm"
+        onItemClick={item => console.log('well done, you clicked the button')}
```
Comment (ciaranschutte) [inline components/pages/file-repository/FileTable/index.tsx:159]: Can we leave out this console.log?

Comment (wajiha-oicr) [reply components/pages/file-repository/FileTable/index.tsx:159]: oh right! that was there from the storybook!! I'll remove it  😬

File: uikit/Table/index.tsx [hunk 4]
```diff
@@ -117,39 +124,93 @@ function Table<Data extends TableDataBase>({
...
+        className={`${className} ${stripped ? '-striped' : ''} ${highlight ? '-highlight' : ''}`}
+        TrComponent={props => (
+          <TrComponent {...props} primaryKey={primaryKey} selectedIds={selectedIds} />
+        )}
+        LoadingComponent={LoadingComponent}
+        getTrProps={(state, rowInfo, column) => ({
+          rowInfo,
+          ...getTrProps(state, rowInfo, column),
+        })}
+        minRows={0}
```
Comment (wajiha-oicr) [inline uikit/Table/index.tsx:202]: lines 203 to line 205

**Pull request closed without merging**